### PR TITLE
Update all the Markdown files for the S24 running of this lab

### LINKED ISSUE #118: Write up something about merge queues
I think merge queues are really useful, and could be particularly useful in this lab where merges and approvals all tend to pile up in confusing ways. So it might make sense to add a little about about merge queues (with links to more complete documentation) in the write-up for this lab.

The GitHub documentation page [Managing a merge queue](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue) explains a lot of the key ideas, but not necessarily in the easiest to follow manner.

### PR DESCRIPTION
**Pull request by NicMcPhee**
There's a lot here, so this is going to be a kind of annoying PR to review. The highlights:

- @kklamberty and I decided that since we now have all these automatic checks in places, the badges aren't really necessary since the green check or red x tells us what we want/need to know. Including and updating the badges does add a little friction, so we'll just remove them.
- I added a lot of detail about how to create a Personal Access Token specifically for this lab so that students can do the command line `git` work described in part one.
- I removed the historical notes about the change to `main` as I think that transition is mostly done now and people are a lot less likely to run into the old approach.
- Added a note about the need to add GitKraken to the organization in the facutly/TA todos. I never remember to do this and get burned at the start of the semester, so I thought it would be good to mention it.
- I attempted to document the new merge queue approach. This is probably the part I'm least confident about, and I may not really know what it looks like until we do it tomorrow in lab.

There were also a bunch of minor grammar fixes, updates, etc. I even found one dangling reference to IntelliJ IDEA, which I switched to VS Code. 😜 

Closes #118

### CODE REVIEW

File: PART_1_SHARE_INFO.md [hunk 1]
```diff
@@ -411,17 +423,63 @@ work up to the GitHub repository:
...
+which means you'll need to create new PATs later in the semester if/when you
+need them. You can go with the defaults for most of the options, but there will
+be some questions/options you'll need to address when creating your PAT:
+
+- You can use whatever **Token name** makes sense to you for the token.
+- I'd go with an **Expiration** of 7 days since we should be done with this lab
+  the end of the week.
+- Under "Resource owner" you _have_ to select the organization for this semester's
+  Software Design course. That's probably something of the form `UMM-CSci-3601-S24`.
+- Under **Reposity access** choose "Only select repositories". Then choose
```
Comment (kklamberty) [inline PART_1_SHARE_INFO.md:460]: typo - should be Repository rather than Reposity

File: PART_2_JAVA_INTRODUCTIONS.md [hunk 2]
```diff
@@ -535,11 +529,11 @@ conflicts, and you'd probably rather use that for more complex conflicts.
 See [our write-up on using GitKraken to deal with merge conflicts](docs/MERGE_CONFLICTS.md)
 for more examples and details.
 
-:warning: :bangbang: At this point Chris (or you if you've been doing something
+:bangbang: At this point Chris (or you if you've been doing something
 similar) will need to pull these changes back down to your copy if you want
 or need to make changes to your branch. The changes you've made by
 resolving conflicts in the web GUI _only_ affect the version of the
-code on GitHub, so you have to pull if you want those changes to be
+code on GitHub, so you have to `pull`` if you want those changes to be
```
Comment (kklamberty) [inline PART_2_JAVA_INTRODUCTIONS.md:536]: extra `?

**Review by kklamberty — CHANGES_REQUESTED** [review]
Tiny typos; otherwise looks great!

**Comment by NicMcPhee** [comment]
Thanks for the careful read through. Both of those are now fixed!

File: PART_1_SHARE_INFO.md (hunk 1 above)
Comment (NicMcPhee) [reply PART_1_SHARE_INFO.md]: And it even had a blue squiggly underline in VS Code. Doh!

**Review by kklamberty — APPROVED** [review]

**Pull request merged**